Sources/Sentry/SentryTracer.m

```diff
@@ -521,16 +538,61 @@ - (BOOL)hasUnfinishedChildSpansToWaitFor
 
 - (void)finishInternal
 {
-    [self cancelDeadlineTimer];
+    BOOL discardTransaction = [self finishTracer];
+    if (discardTransaction) {
+        return;
+    }
+
+    SentryTransaction *transaction = [self toTransaction];
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+    if (self.isProfiling) {
+        NSDate *startTimestamp;
+
+#    if SENTRY_HAS_UIKIT
+        if (appStartMeasurement != nil) {
+            startTimestamp = appStartMeasurement.runtimeInitTimestamp;
+        }
+#    endif // SENTRY_HAS_UIKIT
+
+        if (startTimestamp == nil) {
+            startTimestamp = self.startTimestamp;
+        }
+        if (!SENTRY_ASSERT_RETURN(startTimestamp != nil,
+                @"A transaction with a profile should have a start timestamp already. We will "
+                @"assign the current time but this will be incorrect.")) {
+            startTimestamp = [SentryDependencyContainer.sharedInstance.dateProvider date];
+        }
+
+        sentry_captureTransactionWithProfile(
+            self.hub, self.dispatchQueue, transaction, startTimestamp);
+        return;
+    }
+#endif // SENTRY_TARGET_PROFILING_SUPPORTED
+
+    [_hub captureTransaction:transaction withScope:_hub.scope];
```

Comment:
This code was already here in a previous version. I only moved it to a different method.